src/Test.UnitTests.Sarif/Core/AddressTests.cs

```diff
@@ -0,0 +1,227 @@
+
+namespace Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Core
+{
+    public class AddressTests
+    {
+        private const string AbsoluteAddress = "\"AbsoluteAddress\"";
+        private static readonly TestAssetResourceExtractor s_extractor = new TestAssetResourceExtractor(typeof(AddressTests));
+
+        [Fact]
+        public void Address_VerifyRoundTripFromObject()
+        {
+            // Arrange
+            var testCases = new[] {
+                new { absoluteAddress = (BigInteger?)null, expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)long.MinValue - 1, expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)long.MinValue, expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)int.MinValue - 1, expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)int.MinValue, expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)(-2), expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)(-1), expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)0, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)1, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)2, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)int.MaxValue, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)int.MaxValue + 1, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)long.MaxValue, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)long.MaxValue + 1, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)ulong.MaxValue, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)ulong.MaxValue + 1, expectedShouldSerializeAbsoluteAddress = true },
+            };
+
+            var address = new Address();
+            foreach (var testCase in testCases)
+            {
+                if (testCase.absoluteAddress.HasValue)
+                {
+                    address.AbsoluteAddress = testCase.absoluteAddress.Value;
+                }
+
+                BigInteger expectedAbsoluteAddress =
+                    (!testCase.absoluteAddress.HasValue || testCase.absoluteAddress.Value < 0) ? -1 : testCase.absoluteAddress.Value;
+
+                // Act & Assert
+                Address_VerifyRoundTripFromObjectHelper(address,
+                    expectedShouldSerializeAbsoluteAddress: testCase.expectedShouldSerializeAbsoluteAddress,
```

Comment:
>expectedShouldSerializeAbsoluteAddress [](http://example.com/codeflow?start=20&length=38)

the idea to use this is if you are passing a bool or a magic number, so people would know what is the variable and what it does.

Since the property from the testCase already have the same information, you could remove this.

Reply:
Oh got it, so I will change all from:
expectedShouldSerializeAbsoluteAddress: testCase.expectedShouldSerializeAbsoluteAddress,
to just:
testCase.expectedShouldSerializeAbsoluteAddress,

Reply:
Exactly

Reply:
done